pkg/querier/querier.go

```diff
@@ -456,53 +444,44 @@ func (s *sliceSeriesSet) Warnings() storage.Warnings {
 }
 
 type storeQueryable struct {
-	QueryableWithFilter
-	QueryStoreAfter time.Duration
+	q               Queryable
+	queryStoreAfter time.Duration
 }
 
-func (s storeQueryable) UseQueryable(now time.Time, queryMinT, queryMaxT int64) bool {
-	// Include this store only if mint is within QueryStoreAfter w.r.t current time.
-	if s.QueryStoreAfter != 0 && queryMinT > util.TimeToMillis(now.Add(-s.QueryStoreAfter)) {
-		return false
+func (s storeQueryable) OptionalQuerier(ctx context.Context, now time.Time, queryMinT, queryMaxT int64) (storage.Querier, error) {
+	if s.q == nil {
+		return nil, nil
 	}
-	return s.QueryableWithFilter.UseQueryable(now, queryMinT, queryMaxT)
-}
 
-type alwaysTrueFilterQueryable struct {
-	storage.Queryable
-}
-
-func (alwaysTrueFilterQueryable) UseQueryable(_ time.Time, _, _ int64) bool {
-	return true
-}
-
-// UseAlwaysQueryable wraps storage.Queryable into QueryableWithFilter, with no query filtering.
-func UseAlwaysQueryable(q storage.Queryable) QueryableWithFilter {
-	return alwaysTrueFilterQueryable{Queryable: q}
+	// Include this store only if mint is within QueryStoreAfter w.r.t current time.
+	if s.queryStoreAfter != 0 && queryMinT > util.TimeToMillis(now.Add(-s.queryStoreAfter)) {
+		return nil, nil
+	}
+	return s.q.OptionalQuerier(ctx, now, queryMinT, queryMaxT)
 }
 
 type useBeforeTimestampQueryable struct {
```

Comment:
Is this struct still needed? I can only see it being used in a single test which tests it.